src/connector/postgres/conversion.rs

```diff
@@ -29,6 +29,92 @@ pub fn conv_params<'a>(params: &'a [Value<'a>]) -> Vec<&'a (dyn types::ToSql + S
     params.iter().map(|x| x as &(dyn ToSql + Sync)).collect::<Vec<_>>()
 }
 
+/// Maps a list of query parameter values to a list of Postgres type.
+pub fn params_to_types(params: &[Value<'_>]) -> Vec<PostgresType> {
+    params
+        .iter()
+        .map(|p| -> PostgresType {
+            // While we can infer the underlying type of a null, Prisma can't.
+            // Therefore, we let PG infer the underlying type.
+            if p.is_null() {
+                return PostgresType::UNKNOWN;
+            }
+
+            match p {
+                Value::Int32(_) => PostgresType::INT4,
+                Value::Int64(_) => PostgresType::INT8,
+                Value::Float(_) => PostgresType::FLOAT4,
+                Value::Double(_) => PostgresType::FLOAT8,
+                Value::Text(_) => PostgresType::TEXT,
+                // Enums are special types, we can't statically infer them, so we let PG infer it
+                Value::Enum(_) => PostgresType::UNKNOWN,
+                Value::Bytes(_) => PostgresType::BYTEA,
+                Value::Boolean(_) => PostgresType::BOOL,
+                Value::Char(_) => PostgresType::CHAR,
+                #[cfg(feature = "bigdecimal")]
+                Value::Numeric(_) => PostgresType::NUMERIC,
+                #[cfg(feature = "json")]
+                Value::Json(_) => PostgresType::JSON,
```

Comment:
I'm worried about this here, you should always prefer jsonb if you can, json is only for function return values. Did you have some tests failing if you'd use jsonb here?